sky/backends/backend_utils.py

```diff
@@ -1346,6 +1354,27 @@ def _setup_local_worker(runner: command_runner.SSHCommandRunner):
         subprocess_utils.run_in_parallel(_setup_local_worker, worker_runners)
 
 
+def check_if_ssh_blocked():
+    """ Checks if network blocks outgoing connections to port 22. """
+    # Limit the number of connections to _SSH_TEST_IP.
+    global _SSH_TEST_IP_CHECKED
+    if _SSH_TEST_IP_CHECKED:
+        return
+
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.settimeout(1)  # 1 second timeout
+    try:
+        s.connect((_SSH_TEST_IP, 22))
```

Comment:
<img width="1784" alt="Screen Shot 2022-11-01 at 9 02 30 PM" src="https://user-images.githubusercontent.com/20640975/199392872-66265c61-6c8f-4224-9be4-8c6bc5080688.png">

I tried all other ports on my local home wifi (where sky works), but port 22 seems to not work. Maybe this website does not permit port 22.

Reply:
From portquiz.net:
```
 Please also note that this server uses some port for real services (22, 25), so testing with your browser on those ports will not work. 
```
However, testing port 22 on the command line should work just fine.

Reply:
If `portquiz.net` does not recommend connecting on Port 22, what other websites can be considered for testing connections?

We can just set `_SSH_TEST_IP` to a different domain.